pkg/image/importer/importer.go

```diff
@@ -397,6 +418,7 @@ func importRepositoryFromDocker(ctx gocontext.Context, retriever RepositoryRetri
 			importDigest.Err = err
 			continue
 		}
+		importDigest.Err = calculateImageSize(ctx, repo, importDigest.Image)
```

Comment:
I'm still concerned because each of those stats could fail, resulting in
other failures.  We probably *don't* want to fail import if size can't be
calculated.  We probably don't want to add a fan-out of 20 requests on
setup.

I'd prefer to not fix this until we have schema2 support.  I thought we
were getting a schema2 response without size.

On Wed, Jun 1, 2016 at 2:22 PM, Alexey Gladkov <notifications@github.com>
wrote:

> In pkg/image/importer/importer.go
> <https://github.com/openshift/origin/pull/9115#discussion_r65415556>:
>
> > @@ -397,6 +418,7 @@ func importRepositoryFromDocker(ctx gocontext.Context, retriever RepositoryRetri
> >  			importDigest.Err = err
> >  			continue
> >  		}
> > +		importDigest.Err = calculateImageSize(ctx, repo, importDigest.Image)
>
> it's unnecessarily calculates the size.
>
> @soltysh <https://github.com/soltysh> @smarterclayton
> <https://github.com/smarterclayton> I added the check to calculate the
> size only when it is equal to zero. This would eliminate unnecessary
> requests if we have all the information.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9115/files/492baa03f23cf45ca229c3501c29ca4bccf546f3#r65415556>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p3qVdLRkR2IkXDwpJ7kJ7xgChqOKks5qHc3zgaJpZM4IrzC2>
> .
>